core/jobs.py

```diff
@@ -149,7 +151,13 @@ def enqueue(cls, job_id, queue_name, additional_job_params=None):
                 values.
             additional_job_params: dict(str : *) or None. Additional parameters
                 for the job.
+            test_only: boolean. For testing _real_enqueue method.
         """
+        # For testing the method _real_enqueue.
+        if test_only:
+            cls._real_enqueue(job_id, queue_name, additional_job_params)
```

Comment:
Looking at this in more detail -- could you please explain why you actually need the test_only boolean? In other words, let's say you didn't have it. What prevents you from writing a test that calls enqueue() directly, on a class which has _real_enqueue() not implemented, and catching the exception that results from that?

It would be better to do that if possible, so that you do not write code in the main codebase that is only run in the context of a test (which is an antipattern).

Reply:
@ankita240796 Could you further explain this to me? I assumed that we only want to run tests on _real_enqueue, which resides in the same class as enqueue() - and only called by enqueue(). So I test on enqueue() by only call _real_enqueue without running code any further. In what scenario would enqueue() and _real_enqueue() not reside in the same class? And how does it affect the way I test _real_enqueue()? 
I am sorry for prolonging this PR, I will try to be more careful next time.

Reply:
Ah, no -- you can't run tests only on _real_enqueue() without calling the private method, and the whole point of this issue is to reduce dependency on the public API.

So, instead, write tests that call the entirety of enqueue() instead, such that, as part of the execution of enqueue(), the relevant features of _real_enqueue() also get tested in the process. Does that make sense?

Reply:
So instead of testing _real_enqueue(), I will test on enqueue() only?

Reply:
Yes, that's correct. Thanks!